tap_google_analytics/client.py

```diff
@@ -22,13 +23,41 @@ def is_retryable_403(response):
     https://developers.google.com/analytics/devguides/reporting/metadata/v3/errors
     """
     retryable_errors = {"userRateLimitExceeded", "rateLimitExceeded", "quotaExceeded"}
-    error_reasons = {error.get('reason') for error in response.json().get('error', {}).get('errors',[])}
+    error_reasons = get_error_reasons(response)
 
     if any(error_reasons.intersection(retryable_errors)):
         return True
 
     return False
 
+
+def get_error_reasons(response):
+    """
+    The google apis don't document the way the errors appear in their reponse json in the same way across different api endpoints and versions. This method defensively tries to grab the error reason(s) in all the ways response have shown to have them. Lastly if all those ways fail the error message just shows the full response json.
+    """
+    response_json = response.json()
+
+    error_reasons = set()
+    if 'error' in response_json:
+        error = response_json.get('error')
+        if 'errors' in error:
```

Comment:
I think this boolean needs a type check to ensure it's a dict. If `error` is a string or some other data type, the calls below will fail or give strange results.

Reply:
What happens if this check fails? Maybe it should just fall back to full response?

Reply:
Fair enough I'll add the check.

Reply:
Done